guides/CreateGINPrivateRepo.md

```diff
@@ -0,0 +1,276 @@
+
+  - These placeholder files have the same filenames (and paths) as the corresponding original files, but are instead simply ASCII text files (if the data is unlocked). If we open these placeholder files, we see they contain a path. This path is where the actual content of the corresponding file will be downloaded to, when we request it.
+  - For example, if the placeholder ASCII text file with name `09.08_09.08.2023-01-Left_frame_013230.png` points to this path `/annex/objects/MD5-s15081575--f0a21c00672ab7ed0733951a652d4b49`, it means that when we specifically request for this file's content with `gin get-content 09.08_09.08.2023-01-Left_frame_013230.png`, the actual png file will be downloaded to `.git/annex/objects/Xq/7G/MD5-s15081575--f0a21c00672ab7ed0733951a652d4b49/MD5-s15081575--f0a21c00672ab7ed0733951a652d4b49` (note that the path in the ASCII file and the actual path are somewhat different, since the actual path contains some subdirectories under `objects`). We can actually verify this file is the image by opening it with an image viewer (in mac: `open -a Preview .git/annex/objects/Xq/7G/MD5-s15081575--f0a21c00672ab7ed0733951a652d4b49/MD5-s15081575--f0a21c00672ab7ed0733951a652d4b49`)
+
+- How? Case for a locked dataset
+
+  - When we `gin download` a repository from the GIN server, we get a local "copy" (clone) of the dataset in our machine. It is not strictly a copy, because the large binary files that make up this dataset will only be downloaded as placeholders.
+  - If the data is locked and no content has been downloaded, the symlinks in the working directory will be broken (since there is no data in the git annex to retrieve).
+  - To get the actual content in the git annex, we need to run `gin download --content`. This will fetch the content from the GIN server (presumably). After this, the symlinks in the working directory should work
+
+- And in an existing directory?
+
+  - After initialising the GIN repo in the current directory and adding a remote, we would commit the data. When committing, the data is "copied" to the git annex.
+  - To replace the copies in the working directory with symlinks to the git annex content, we run `gin lock <path-to-data>`
+  - If we commit this state change and upload the changes to the GIN server, the files will be locked for any future clones of the repo.
+
+- Useful tools for inspecting this further
+
+  - `file` shows the type of file (inspecting the file, rather than plainly looking at the extension like Finder does)
+  - `open -a Preview <>` to open a png file that has no extension
+  - `ls -l <path to symlink` to check the path a symlink points to
+
+- Pending questions
+  - What is the best approach re this "data duplication"
+  - Where is the content of the files stored? Is it in the GIN server?
```

Comment:
The answer is yes (as of your experiments today), right?
And then this PR could "close" #79.